src/AsmResolver.DotNet/Builder/PEImageBuildContext.cs

```diff
@@ -12,24 +12,43 @@ public class PEImageBuildContext
         /// </summary>
         public PEImageBuildContext()
         {
-            DiagnosticBag = new DiagnosticBag();
-        }
-        
+            ErrorListener = new DiagnosticBag();
+        }
+
         /// <summary>
         /// Creates a new build context.
         /// </summary>
-        /// <param name="diagnosticBag">The diagnostic bag to use.</param>
+        /// <param name="diagnosticBag">The diagnostic bag to use.</param>
+        [Obsolete]
         public PEImageBuildContext(DiagnosticBag diagnosticBag)
         {
-            DiagnosticBag = diagnosticBag ?? throw new ArgumentNullException(nameof(diagnosticBag));
-        }
-        
+            ErrorListener = diagnosticBag ?? throw new ArgumentNullException(nameof(diagnosticBag));
+        }

```

Comment:
Maybe mention in the `Obsolete` attribute what the new preferred constructor is.